File: src/com/riadsafowan/java/Practice3.java
package com.riadsafowan.java;

import java.util.Scanner;

public class Practice3 {
    //   1. Write a java program to print the factorial of a number by defining a method named 'Factorial'.
    static class Task1 {
        public static void main(String[] args) {
            Scanner scanner = new Scanner(System.in);
            System.out.print("Enter the number : ");
            int num = scanner.nextInt();
            System.out.println("Factorial of " + num + " is " + factorial(num));
        }

        private static double factorial(int num) {
            long factorial = 1;
            for (int i = 1; i <= num; i++) {
                factorial = factorial * i;
            }
            return factorial;
        }
    }

    static class Task2 {
        //   2. Write a method named isPrime, which takes an integer as an argument and returns true
        //   if the argument is a prime number, or false otherwise.
        //   Also write main method that displays prime numbers between 1 to 500.
        public static void main(String[] args) {

            for (int i = 1; i <= 500; i++) {
                if (isPrime(i))
                    System.out.print(i + ", ");
            }
        }

        private static boolean isPrime(int num) {
            for (int i = 2; i < num; i++) {
                if (num % i == 0) {
                    return false;
                }
            }
            return true;
        }
    }
}

File: src/com/riadsafowan/kotlin/Practice3.kt
package com.riadsafowan.kotlin

import java.util.*

class Practice3 {
    //   1. Write a java program to print the factorial of a number by defining a method named 'Factorial'.
    internal object Task1 {
        @JvmStatic
        fun main(args: Array<String>) {
            val scanner = Scanner(System.`in`)
            print("Enter the number : ")
            val num = scanner.nextInt()
            println("Factorial of " + num + " is " + factorial(num))
        }

        private fun factorial(num: Int): Long {
            var factorial: Long = 1
            for (i in 1..num) {
                factorial *= i
            }
            return factorial
        }
    }

    internal object Task2 {
        //   2. Write a method named isPrime, which takes an integer as an argument and returns true
        //   if the argument is a prime number, or false otherwise.
        //   Also write main method that displays prime numbers between 1 and 500.
        @JvmStatic
        fun main(args: Array<String>) {
            for (i in 1..500) {
                if (isPrime(i)) print("$i, ")
            }
        }

        private fun isPrime(num: Int): Boolean {
            for (i in 2 until num) {
                if (num % i == 0) {
                    return false
                }
            }
            return true
        }
    }

}